src/adhocracy_core/adhocracy_core/websockets/server.py

```diff
@@ -51,6 +51,13 @@ def subscribe(self, client: Hashable, resource: IResource) -> bool:
         if self.is_subscribed(client, resource):
             return False
         path = resource_path(resource)
+
+        # ensure that no private info leaks to the outside
+        if path.startswith('/principals/') or path.startswith('/catalogs/'):
+            logger.info('Unallowed subscribtion')
+            raise WebSocketError('forbidden', 'You are not allowed to '
+                                 'subscribe on {}'.format(path))
+
```

Comment:
we use colander vor client data validation,  so we should add a validator to 
adhocracy_core/websockets/schemas.ClientRequestSchema field 'resource' 

Apart from validating subscriptions I would go a bit deeper. 
websockets.client is responsible for notifing the websocket server. That means it is also responsible to filter what changelog event are send to the websocket server, and event regarding /principals/.. should not be sent at all to the outside world.


Style note: the comment is needless if you use good variables names, for example::

     forbidden_paths =  ('/principals', '/catalogs')
     if  path.starts_with(forbidden_paths):
     ...

test are missing